Feat/update promethus stack 426

smerlos opened the pull request:
# Summary of Changes

This PR introduces the following updates and improvements:

## Fixes

 - CI Configuration: Corrected image repository URL for the end-to-end testing step in Drone CI.

## Features

Documentation:
 - Updated the compatibility matrix and image tags for:
 - node-exporter
 - kube-rbac-proxy
 - prometheus-adapter
 - Added release notes for version v3.3.0.
###  Drone CI:
 - Updated the Kubernetes target version to v1.31.0.
 - Added a new pipeline for Kubernetes version v1.31.1.
###  Grafana:
 - Updated the Grafana image tag to 11.3.0 and added license information to dashboard sources patches.
###  Prometheus Operator:
 - Updated PrometheusRule CRD with new version and enhanced descriptions.
 - Upgraded Prometheus Operator to version 0.76.2
 
###  Kubernetes Monitoring:

 - Moved CPUThrottlingHigh alert configuration to configs/kubeadm.
 - Updated and refined KubeControllerManager alert descriptions.
 - Moved deprecated KubeClientCertificateExpiration alerts.
 
###  Prometheus:
 - Upgraded Prometheus version to 2.54.1.
 - Updated related configurations.
 - Introduced  `scrapeConfigNamespaceSelector: {}`
 - Introduced `scrapeConfigSelector: {}`
 - remove unused scrape config 
 
###  Prometheus Adapter:
 - Upgraded to version 0.12.0.
 - Updated related configurations.

###  Node Exporter:
 - Upgraded to version 1.8.2.
 - Updated related configurations.
 
###  Kube-State-Metrics:
 - Upgraded to version 2.13.0.
 - Updated related images.

###  Blackbox Exporter:
 - Updated image tags and added configmap-reload image.
 - Upgraded Blackbox Exporter to version 0.25.0.

### HAProxy
- Remove HaproxyHttpSlowingDown alert rule

## Testing Performed

1. The following tests were conducted to ensure functionality and stability:

### Cluster Testing:
Deployed changes and conducted tests on Kubernetes clusters using kind versions:
 - v1.29
 - v1.30
 - v1.31
 - Verified backward compatibility with previous module versions.
2. Workload Validation:
 - Confirmed that all workloads run successfully across all tested Kubernetes versions.
3. Dashboard Validation:
 - Verified functionality of all dashboards in Grafana.
 - Identified an issue with the Networking Dashboard (Typha)
4. The upgrade process from version 3.2.0 to this one was tested without issues. the testing was conducted in versions  (1.29, 1.30, and 1.31) using a kind cluster. 

Notes

 Additional investigation is required to address the issue with the Typha dashboard.

ralgozino commented on katalog/prometheus-operated/prometheus.yaml:
```diff
@@ -42,11 +42,13 @@ spec:
       memory: 400Mi
   ruleNamespaceSelector: {}
   ruleSelector: {}
+  scrapeConfigNamespaceSelector: {}
+  scrapeConfigSelector: {}
```
we should mention that we are enabling this in the release notes too

ralgozino commented on katalog/prometheus-operator/kustomization.yaml:
```diff
@@ -10,10 +10,10 @@ namespace: monitoring
   - name: quay.io/prometheus-operator/prometheus-operator
     newName: registry.sighup.io/fury/prometheus-operator/prometheus-operator
-    newTag: v0.67.1
+    newTag: v0.76.2
```
```suggestion
```

ralgozino commented on katalog/prometheus-operator/kustomization.yaml:
```diff
@@ -10,10 +10,10 @@ namespace: monitoring
   - name: quay.io/brancz/kube-rbac-proxy
     newName: registry.sighup.io/fury/brancz/kube-rbac-proxy
-    newTag: v0.14.2
+    newTag: v0.18.1
```
```suggestion
```

ralgozino reviewed (changes requested):
overall great job! some minor fixes and some questions:
1. Did you test the upgrade from the previous version to this one? Anything worth mentioning in the release notes regarding the upgrade process?
2. I don't see any of the MAINTENANCE.md files updated, were all OK? 
3. Please add a note to Grafana's maintenance guide saying that we changed the image tag because of the "unauthenticated" bug we found, so it is expected to not match upstream.

smerlos commented on docs/COMPATIBILITY_MATRIX.md:
```diff
@@ -1,17 +1,18 @@
+| v3.0.1                              |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: |                    |                    |                   |                   |
+| v3.0.2                              |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: |                    |                    |                   |                   |
+| v3.1.0                              |                    |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: |:white_check_mark: |:white_check_mark: |
+| v3.2.0                              |                    |                    |                    |                    |                    |                    |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: |:white_check_mark: |:white_check_mark: |
```
fixed! thanks for noted it 🤟

smerlos commented on katalog/node-exporter/kustomization.yaml:
```diff
@@ -14,10 +14,10 @@ images:
   - name: quay.io/prometheus/node-exporter
     newName: registry.sighup.io/fury/prometheus/node-exporter
-    newTag: v1.6.1
+    newTag: v1.8.2
```
in this case, the upstream suggests the latest, and we are forcing the exact tag.

smerlos commented on katalog/kube-proxy-metrics/kustomization.yaml:
```diff
@@ -11,7 +11,7 @@ namespace: monitoring
 - name: kube-rbac-proxy
   newName: registry.sighup.io/fury/brancz/kube-rbac-proxy
-  newTag: v0.14.0
+  newTag: v0.18.1
```
resolved

smerlos commented on katalog/kube-state-metrics/kustomization.yaml:
```diff
@@ -11,10 +11,10 @@ namespace: monitoring
   - name: quay.io/brancz/kube-rbac-proxy
     newName: registry.sighup.io/fury/brancz/kube-rbac-proxy
-    newTag: v0.14.2
+    newTag: v0.18.1
```
resolved

smerlos commented:
> overall great job! some minor fixes and some questions:
> 
> 1. Did you test the upgrade from the previous version to this one? Anything worth mentioning in the release notes regarding the upgrade process?
> 2. I don't see any of the MAINTENANCE.md files updated, were all OK?
> 3. Please add a note to Grafana's maintenance guide saying that we changed the image tag because of the "unauthenticated" bug we found, so it is expected to not match upstream.

1. at this moment, for this pr, the upgrade process was smooth no important actions are required.
2. overall, are all ok I've just updated the script to add minor changes and also mentioned in the grafana maintenance guide the custom tag for grafana in this release
3. the note was added

Thanks for your comments all of them were addressed correctly

ralgozino reviewed (approved):
LGTM 🚀
